Library/Homebrew/formula_installer.rb

```diff
@@ -942,13 +943,25 @@ def post_install
     @show_summary_heading = true
   end
 
-  def pour
+  def fetch_and_build
+    compute_dependencies unless ignore_deps?
+    pour_bottle? ? fetch_bottle : build
+
+    @fetched = true
```

Comment:
I think this might be better:
```suggestion
    if pour_bottle?
      fetch_bottle
      @fetched = true
    else
      build
      @built = true
    end
```

Basically saying if it's a bottle, fetch it and set `@fetched` to true. If it's not a bottle, build it and set `@built` to true.

Either way, later in the process when `FormulaInstaller#install` is called we won't re-download or re-build.